src/quacc/calculators/espresso/calculators.py

```diff
@@ -0,0 +1,127 @@
+from quacc import SETTINGS
+from quacc.calculators.espresso.core import EspressoCalculator
+from quacc.calculators.espresso.keys import (DYNMAT_KEYS, NEB_KEYS, PH_KEYS,
+                                             Q2R_KEYS, MATDYN_KEYS, PW_KEYS)
+from quacc.calculators.espresso.utils import parse_pw_preset
+
+# Mini classes that might be useful for binaries
+# that have very specific requirements
+
+class Ph(EspressoCalculator):
+    # These are not supposed to change
+    _BIN = 'ph'
+    _KEYS = PH_KEYS
+    _CMD = SETTINGS['ESPRESSO_PH_CMD']
+
+    def __init__(self, input_data):
+        self.input_data = input_data
+        super().__init__()
+
+class Neb(EspressoCalculator):
+
+    # I think it makes sense to have the defaults
+    # here. They are not supposed to change.
+    # Users can have a look at them by doing:
+    # from quacc.calculators.espresso.calculators import Neb
+    # Neb._calc_defaults without looking at the source code
+
+    # Note to self: 'When a child class method is created with the same name and signature as one in the parent, the child's method takes precedence' didn't know that, praise python
+    _BIN = 'neb'
+    _KEYS = NEB_KEYS
+    _CMD = SETTINGS['ESPRESSO_NEB_CMD']
+
+    _calc_defaults = {
+        'input_data': {
+            'path': {
+                'string_method': 'neb',
+                'restart_mode': 'from_scratch',
+                'opt_scheme': 'broyden',
+                'CI_scheme': 'no-CI',
+                'first_last_opt': False,
+                "num_of_images": 6
+            }
+        }
+    }
+
+    def __init__(self, neb_input_data, pw_input_data):
+        self.input_data = neb_input_data
+        self.additional_data = pw_input_data
+        super().__init__()
+        self.input_file = 'neb.dat'
+
+    # It would be nice to be able to run sanity_checks BEFORE
+    # the calculation is submitted to any kind of HPC
+    # This could also raise warnings to remind the user
+    # about known problems i.e. if first_last_opt is not True
+    # and the first and last images contains spin state. Running
+    # the first singlepoint with fresh wavefunctions might end up
+    # in a different spin state, screwing up the whole NEB calculation.
```

Comment:
There are cases where one wants to run an NEB calculation where the initial and final spin states differ. I do this quite frequently, e.g. during spin-crossover reactions or other multi-state reactivity reaction coordinates. So, I'd be hesitant about enforcing the same spin manifold here (although a sanity check to avoid _accidental_ scenarios with the wavefunctions does seem useful).

Reply:
It happens that I optimise the initial and final state to low forces and then when I launch the NEB calculation one/both of them suddenly have high forces due to a different spin state. What I means here is that if we detect that initial/final states have spin and that 'first_last_opt' is not true (meaning that NEB will consider initial and final state as optimised and freeze them) we print a warning, no action taken.

Reply:
Ah, makes sense! Thanks for clarifying!